purchase_request_to_procurement/models/purchase_request.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseRequest(models.Model):
+
+    _inherit = 'purchase.request'
+
+    @api.onchange('warehouse_id')
+    def onchange_warehouse_id(self):
+        """
+        When changing the warehouse: take the related lot_stock_id just
+        like Odoo does into stock.procurement
+        """
+        if self.warehouse_id:
+            self.location_id = self.warehouse_id.lot_stock_id.id
+
+    location_id = fields.Many2one(
+        comodel_name='stock.location', string='Location')
+    warehouse_id = fields.Many2one(
```

Comment:
I just made the the same as the procurement behavior form Odoo I think it is better for consistency between both